### PR TITLE
Support `system.model-name` hardware requirement for `mrack`

### PR DESCRIPTION
**Pull request by skycastlelily**
Pull Request Checklist

* [x] implement the feature
* [x] write the documentation
* [x] extend the test coverage
* [x] update the specification
* [ ] adjust plugin docstring
* [ ] modify the json schema
* [ ] mention the version
* [x] include a release note

### CODE REVIEW

**Review by happz — APPROVED** [review]

File: tmt/steps/provision/mrack.py [hunk 1]
```diff
@@ -567,6 +567,23 @@ def _transform_system_numa_nodes(
         children=[MrackHWBinOp('numanodes', beaker_operator, actual_value)])
 
 
+def _transform_system_model_name(
+        constraint: tmt.hardware.NumberConstraint,
```
Comment (therazix) [inline tmt/steps/provision/mrack.py:571]: Shouldn't it be `TextConstraint`?
```suggestion
        constraint: tmt.hardware.TextConstraint,
```

**Comment by skycastlelily** [comment]
Ah,yeah😅, thanks  for your review,and updated:)

**Comment by therazix** [comment]
@skycastlelily I believe you accidentally edited the `_transform_cpu_model` function. I meant changing the `NumberConstraint` to `TextConstraint` in the `_transform_system_model_name` function.

**Comment by skycastlelily** [comment]
ah,yes😅,thanks a lot for pointing that out,updated,should be good now:)

**Review by therazix — APPROVED** [review]

**Pull request merged**